addons/l10n_no_edi/models/account_edi_format.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+
+from odoo import models, _
+from odoo.addons.account_edi_ubl_bis3.models.account_edi_format import COUNTRY_EAS
+
+import base64
+import markupsafe
+
+
+class AccountEdiFormat(models.Model):
+    _inherit = 'account.edi.format'
+
+    ####################################################
+    # Import
+    ####################################################
+
+    def _is_ubl(self, filename, tree):
+        """ OVERRIDE so that the generic ubl parser does not parse BIS3 any longer.
+        """
+        is_ubl = super()._is_ubl(filename, tree)
+        return is_ubl and not self._is_ehf_3(filename, tree)
+
+    def _is_ehf_3(self, filename, tree):
+        profile_id = tree.find('./{*}ProfileID')
+        nor_tax = tree.xpath("//*[text()='Foretaksregisteret']")
+        return tree.tag == '{urn:oasis:names:specification:ubl:schema:xsd:Invoice-2}Invoice' \
+            and profile_id is not None \
+            and 'peppol' in profile_id.text \
+            and nor_tax is not None
+
+    def _bis3_get_extra_partner_domains(self, tree):
+        if self.code == 'ehf_3':
+            endpoint = tree.find('./{*}AccountingSupplierParty/{*}Party/{*}EndpointID')
+            if endpoint is not None:
+                scheme = endpoint.attrib['schemeID']
+                if scheme == '0192' and endpoint.text:
```

Comment:
FWIW using xpath the entire thing should be done at once e.g.
```python
tree.xpath('./cac:AccountingCustomerParty/cac:Party/cbc:EndpointId[@schemeId="0192"]/text()', namespaces=peppol_ns)
```
or something along those lines.